### PR TITLE
Refactor signup/signin to service

### PR DESCRIPTION
**Pull request by darenang44**

### CODE REVIEW

**Comment by underaj** [comment]
LGTM

File: client/app/components/auth/auth.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ angular.module('throwcast.auth', ['ngRoute'])
 .config(function ($routeProvider) {
   $routeProvider
   .when('/signin', {
-      templateUrl: 'app/auth/signin.html',
+      templateUrl: 'app/components/auth/signin.html',
       controller: 'AuthController'
   })
```
Comment (underaj) [inline client/app/components/auth/auth.js:8]: spaces

File: client/app/components/auth/auth.service.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    signin: function (userCredentials) {
+      return $http.post('http://localhost:8888/auth/local/', userCredentials)
+      .then(function (res) {
+        $window.localStorage.token = res.data.token;
+        $location.path('/');
+      })
+      .catch(function (res) {
+        delete $window.localStorage.token;
+      });
+    },
```
Comment (dting) [inline client/app/components/auth/auth.service.js:24]:
```
signUp: function(userCredentials) {
  return userService.createUser(userCredentials)
    .then(function(res) {     
       $window.localStorage.token = res.data.token;
    })
    .catch(function () {
      delete $window.localStorage.token;
    });
},
```

File: client/app/components/auth/auth.controller.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+angular.module('throwcast.auth')
+
+.controller('AuthController', function ($scope, authService, userService) {
+  $scope.userCredentials = {username: '', password: ''};
+  $scope.createUser = userService.createUser;
+  $scope.signin = authService.signin;
```
Comment (dting) [inline client/app/components/auth/auth.controller.js]:
```
$scope.signIn = function() {
  authService.signIn($scope.userCredentials)
    .then(function() {
      $window.redirect('/');
    })
    .catch(function(err) {
       // handle error;
      $scope.error = 'Error :' + err
    });
};
```

File: client/app/components/auth/auth.controller.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+angular.module('throwcast.auth')
+
+.controller('AuthController', function ($scope, authService, userService) {
+  $scope.userCredentials = {username: '', password: ''};
+  $scope.createUser = userService.createUser;
```
Comment (dting) [inline client/app/components/auth/auth.controller.js]:
```
$scope.signUp = function() {
  authService.signUp($scope.userCredentials)
    .then(function() {
      $window.redirect('/');
    })
    .catch(function(err) {
       // handle error;
      $scope.error = 'Error :' + err
    });
}
```

File: client/app/components/auth/auth.controller.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+angular.module('throwcast.auth')
+
+.controller('AuthController', function ($scope, $location, authService, userService) {
+  $scope.userCredentials = {username: '', password: ''};
+  $scope.createUser = userService.createUser;
```
Comment (dting) [inline client/app/components/auth/auth.controller.js]: This doesn't need to be here.

File: client/app/components/auth/auth.controller.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+angular.module('throwcast.auth')
+
+.controller('AuthController', function ($scope, $location, authService, userService) {
```
Comment (dting) [inline client/app/components/auth/auth.controller.js]: Remove the user service injection

File: client/app/components/auth/auth.service.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+angular.module('throwcast.auth')
+
+.factory('authService', function ($http, $window, $location, userService) {
+  return {
+    signUp: function (userCredentials) {
+      return userService.createUser(userCredentials)
+      .then(function (res) {
+       $window.localStorage.token = res.data.token;
```
Comment (dting) [inline client/app/components/auth/auth.service.js]: indentation

File: client/app/components/auth/auth.service.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      })
+      .catch(function () {
+      delete $window.localStorage.token;
```
Comment (dting) [inline client/app/components/auth/auth.service.js]: indentation

File: client/app/components/auth/user.service.js [hunk 9]
```diff
@@ -1,46 +1,45 @@
...
   var message;
   return {
+    createUser: function (userCredentials) {
+      return $http.post('http://localhost:8888/api/users/', userCredentials);
+    },
     getUser: function () {
       return $http.get('http://localhost:8888/users/me')
       .then(function (res) {
-        user.data = res.data;
+        user = res.data;
```
Comment (dting) [inline client/app/components/auth/user.service.js]: This reassigns the user variable. Any service that has grabbed the user object from `userService.user` will have a stale reference. You should be changing the properties of user no doing a reassignment.

File: client/app/pod/popular.controller.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
 angular.module('throwcast.popular')
 
-.controller('PopularController', function ($scope, $http) {
+.controller('PopularController', function ($scope, $http, userService) {
```
Comment (dting) [inline client/app/pod/popular.controller.js]: You aren't using the `userService` right now. Remove this injection until it is needed.

File: client/app/components/auth/user.service.js [hunk 11]
```diff
@@ -1,46 +1,45 @@
 angular.module('throwcast.auth')
 
-.factory('userService', function ($http) {
+.factory('userService', function ($http, $window, $location) {
```
Comment (dting) [inline client/app/components/auth/user.service.js]: You aren't using $window or $location in this service. Remove the injections.

File: client/app/components/auth/signin.html [hunk 12]
```diff
@@ -1,14 +1,14 @@
...
-    <input type="text" ng-model="user.username">
+    <input type="text" ng-model="userCredentials.username">
     <br>
     Password:
     <br>
-    <input type="text" ng-model="user.password">
+    <input type="text" ng-model="userCredentials.password">
     <br>
-    <button ng-click="signin()">Signin</button>
+    <button ng-click="signIn(userCredentials)">SignIn</button>
```
Comment (dting) [inline client/app/components/auth/signin.html]: This `signIn` function doesn't take an argument. It uses `$scope.userCredentials` internally.

File: client/app/components/auth/signup.html [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+  <h1>Sign Up</h1>
+    Username:
+    <br>
+    <input type='text' ng-model="userCredentials.username">
+    <br>
+    Password:
+    <br>
+    <input type="text" ng-model="userCredentials.password">
+    <br>
+    <button ng-click="signUp(userCredentials)">SignUp</button>
```
Comment (dting) [inline client/app/components/auth/signup.html]: This `signUp` function doesn't take an argument. It uses `$scope.userCredentials` internally.

File: client/app/components/auth/auth.controller.js (hunk 5 above)
Comment (darenang44) [inline client/app/components/auth/auth.controller.js]: Done

File: client/app/components/auth/auth.controller.js (hunk 6 above)
Comment (darenang44) [inline client/app/components/auth/auth.controller.js]: Done

File: client/app/components/auth/auth.service.js (hunk 7 above)
Comment (darenang44) [inline client/app/components/auth/auth.service.js]: Done

File: client/app/components/auth/auth.service.js (hunk 8 above)
Comment (darenang44) [inline client/app/components/auth/auth.service.js]: Done

File: client/app/components/auth/user.service.js (hunk 9 above)
Comment (darenang44) [inline client/app/components/auth/user.service.js]: Done

File: client/app/pod/popular.controller.js (hunk 10 above)
Comment (darenang44) [inline client/app/pod/popular.controller.js]: Done

File: client/app/components/auth/user.service.js (hunk 11 above)
Comment (darenang44) [inline client/app/components/auth/user.service.js]: Done

File: client/app/components/auth/signin.html (hunk 12 above)
Comment (darenang44) [inline client/app/components/auth/signin.html]: Done

File: client/app/components/auth/signup.html (hunk 13 above)
Comment (darenang44) [inline client/app/components/auth/signup.html]: Done

**Comment by dting** [comment]
LGTM

**Pull request merged**